pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java

```diff
@@ -299,4 +296,189 @@ public void testNullLiteralSelectionOnlyBroker()
     assertEquals(rows.size(), 1);
     assertEquals(rows.get(0).get(0).asText(), "null");
   }
+
+  @Test
+  public void testNullSelectionOnlyTransform()
```

Comment:
nit: break down the test into multiple tests so that we test behaviors separately.